src/Utils.cc

```diff
@@ -60,15 +60,36 @@ bool loadPose(sdf::ElementPtr _sdf, ignition::math::Pose3d &_pose,
   std::pair<std::string, bool> framePair =
       sdf->Get<std::string>("relative_to", "");
 
+  // Read the rotation type attribute.
+  std::pair<std::string, bool> rotationTypePair =
+      sdf->Get<std::string>("rotation_type", "rpy_radians");
+
   // Read the pose value.
   std::pair<ignition::math::Pose3d, bool> posePair =
     sdf->Get<ignition::math::Pose3d>("", ignition::math::Pose3d::Zero);
 
   // Set output, but only if the return value is true.
   if (posePair.second)
   {
-    _pose = posePair.first;
     _frame = framePair.first;
+
+    // Modify the rotation component based on the rotation type expected.
+    // Only rpy_degrees require a conversion from degrees to radians.
+    if (rotationTypePair.second && rotationTypePair.first == "rpy_degrees")
```

Comment:
It's very odd to transform the representation after constructing the semantic value for it.

Can you make sure the parsing function receives `@rotation_type` in some form or fashion?

Reply:
Added a function to pass element attributes during value parsing, https://github.com/ignitionrobotics/sdformat/pull/589/commits/aac7829b64c45ab18f667c715e57c4e0f34fa5cf#diff-c4791194513743d2432582e6f31c1c41d3959b1505edff9114e50396156064c9R187